torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -1840,6 +1840,29 @@ def _(func, types, args, kwargs):
             weight_tensor = weight_tensor.dequantize()
         return torch.nn.functional.linear(input_tensor, weight_tensor, bias)
 
+@implements(torch.nn.functional.embedding)
+def _(func, types, args, kwargs):
+    # new_arg1 = args[1].dequantize()
+    idx = args[0]
+    int_data, scale, zero_point = args[1].tensor_impl.get_plain()
+    assert kwargs["padding_idx"] is None and kwargs["max_norm"] is None and not kwargs["scale_grad_by_freq"] and not kwargs["sparse"] and kwargs["norm_type"]==2.0
+    sliced_data, sliced_scale, sliced_zero_point = int_data[idx], scale[idx], zero_point[idx]
```

Comment:
is there any restrictions on `idx` for this to be valid?

Reply:
not as far as our tests show